pkg/controllers/management/auth/globalroles/globalrolebinding_handler.go

```diff
@@ -137,6 +162,142 @@ func (grb *globalRoleBindingLifecycle) deleteAdminBinding(obj *v3.GlobalRoleBind
 	return nil
 }
 
+// reconcileClusterPermissions grants permissions for the binding in all downstream (non-local) clusters. Will also
+// remove invalid bindings (bindings not for active RoleTemplates or for invalid subjects).
+func (grb *globalRoleBindingLifecycle) reconcileClusterPermissions(globalRoleBinding *v3.GlobalRoleBinding) error {
+	globalRole, err := grb.grLister.Get("", globalRoleBinding.GlobalRoleName)
+	if err != nil {
+		return fmt.Errorf("unable to get globalRole %s: %w", globalRoleBinding.Name, err)
+	}
+	clusters, err := grb.clusterLister.List("", labels.Everything())
+	if err != nil {
+		return fmt.Errorf("unable to list clusters when reconciling globalRoleBinding %s: %w", globalRoleBinding.Name, err)
+	}
+
+	var missedClusters bool
+	for _, cluster := range clusters {
+		// we don't sync permissions for the local cluster, but we do want to purge user-created permissions
+		if cluster.Name == localClusterName {
+			// inheritedClusterRoles only apply on non-local clusters, so skip the local cluster
+			err := grb.purgeCorruptRoles([]string{}, cluster, globalRoleBinding)
+			if err != nil {
+				// failure to remove bad bindings shouldn't affect our ability to sync new permissions, so we log and keep processing
+				logrus.Errorf("unable to purge roles for cluster %s and grb %s, some bindings may remain: %s", cluster.Name, globalRoleBinding.Name, err.Error())
+			}
+			continue
+		}
+		err := grb.purgeCorruptRoles(globalRole.InheritedClusterRoles, cluster, globalRoleBinding)
+		if err != nil {
+			// failure to remove bad bindings shouldn't affect our ability to sync new permissions, so we log and keep processing
+			logrus.Errorf("unable to purge roles for cluster %s and grb %s, some bindings may remain: %s", cluster.Name, globalRoleBinding.Name, err.Error())
```

Comment:
Are we okay with not re-enqueuing this GRB when we fail to clean up the old CRTBs?
```suggestion
			logrus.Errorf("unable to purge roles for cluster %s and grb %s, some bindings may remain: %s", cluster.Name, globalRoleBinding.Name, err.Error())
			missedCluster = true
```

Reply:
Per offline discussion, changed this so that we do re-enqueue if we fail to clean up the old crtbs.